File: bubble_sort/BubbleSort.cs
public class Bubble_Sort
{
    static void Main(string[] args)
    {
        int[] arr = { 89, 2, 24, 1002, 4,1,8 };
        int[] sortedArray = BubbleSort(arr);

        foreach(int item in sortedArray)
        {
            Console.WriteLine(item);
        }
        Console.ReadKey();
    }

    /// <summary>
    /// Bubble Sort
    /// </summary>
    /// <param name="arr">Array for sorting</param>
    /// <returns>Sorted array</returns>
    /// <exception cref="Exception">Raise exception if array is null</exception>
    static int[] BubbleSort(int[] arr) { 

        if(arr == null || arr.Length == 0)
        {
            throw new Exception("Passed Array is Invalid");
        }

        if(arr.Length == 1)
        {
            return arr;
        }

        int temp = 0;
        int length = arr.Length;
        bool swapNeeded;

        for (int i = 0 ; i < length; i++) {
            swapNeeded = false;
            
            for (int j = 0 ; j < length - i - 1; j++) {

                if (arr[j] > arr[j + 1])
                {
                    temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swapNeeded = true;
                }
            
            }

            if (!swapNeeded)
            {
                break;
            }
        
        }
        return arr;
    
    
    }
}

File: bubble_sort/bubblesort.rb
def bubble_sort(arr)
    arr_len = arr.length
    (0...arr_len - 1).each do |i|
      (0...arr_len - i - 1).each do |j|
        if arr[j] > arr[j + 1]
          arr[j], arr[j + 1] = arr[j + 1], arr[j]
        end
      end
    end
  end
  
  # Example array
  arr = [99, 47, 86, 64, 25, 15, 120,2,67,92,73]
  
  # Call the bubble_sort function to sort the array
  bubble_sort(arr)
  
  # Print the sorted array
  puts "The sorted array is:"
  arr.each do |num|
    puts num
  end
  